engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -22,10 +28,22 @@ class EngineJobExecutionActor(jobKey: BackendJobDescriptorKey,
                               restarting: Boolean,
                               serviceRegistryActor: ActorRef,
                               jobStoreActor: ActorRef,
+                              backendName: String,
                               callCachingMode: CallCachingMode) extends LoggingFSM[EngineJobExecutionActorState, EJEAData] with WorkflowLogging {
 
   override val workflowId = executionData.workflowDescriptor.id
 
+  val tag = s"EJEA [UUID(${workflowId.shortString}/${jobKey.call.fullyQualifiedName}:${jobKey.index.fromIndex}(attempt ${jobKey.attempt}))]"
+
+  // There's no need to check for a cache hit again if we got preempted
+  // NB: this can also change (e.g. if we have a HashError we just force this to CallCachingOff)
+  private var effectiveCallCachingMode = if (jobKey.attempt > 1) { callCachingMode.withoutRead } else { callCachingMode }
```

Comment:
I'm kind of curious why the king of FSM love chose to manage state via a var in this one case